tests/core/consensus/test_pow_mining.py

```diff
@@ -80,7 +82,7 @@ class ChainClass(MiningChain):
 
     chain = genesis(ChainClass)
 
-    block = chain.mine_block(difficulty=1)
+    block = chain.mine_block(difficulty=3)
```

Comment:
any important context to leave behind in comments for `TEST_NUM_CACHES`, difficulty number, or `_concurrently_run_to_completion` numbers or is that sufficiently summarized by the `skip` reasons?